### ISSUE TITLE
want to be able to set ExcelDataReader.SheetNames PUBLIC

### ISSUE THREAD

**Issue by lileyzhao** [MarkPflug/Sylvan.Data.Excel#87]
For some usage scenarios, I want to be able to set the access scope for SheetInfo[] sheetNames in ExcelDataReader to public,

Now I can only use the source code to get SheetInfo[]

**Comment by MarkPflug** [comment]
> For some usage scenarios

Can you describe the scenario?

**Comment by MarkPflug** [comment]
This is a reasonable request, but I'll have to give it some thought. I don't want to just make the sheet names public, because I feel like for this to be useful, a user would also want to "open" each specific sheet. Currently the "NextResult" is the way to do this, but that only allows moving forward through the sheets. If someone opened a specific sheet, what should happen when NextResult is called? Presumably, it would move to the next sheet.

For now, you can open the sheet twice: the first time through call NextResult and WorksheetName to get the ordered list of worksheets. Then, close and reopen the file and use the sheetnames to provide the user options.

**Comment by lileyzhao** [comment]
You're right, we're more of a financial business, where multiple pieces of data are mixed into one file through different worksheets, so you need to specify a specific sheet for import.



It is possible to open the sheet twice. I can work like this for a while.

**Comment by lileyzhao** [comment]
@MarkPflug Hello, do you have a plan to disclose the `ExcelDataReader.SheetNames`

**Comment by lileyzhao** [comment]
Wow, finally support. Thank you very, very much
